src/main/java/com/hpe/adm/octane/ideplugins/services/connection/ClientLoginCookie.java

```diff
@@ -0,0 +1,90 @@
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.net.HttpCookie;
+import java.util.List;
+import java.util.Optional;
+
+
+public class ClientLoginCookie {
+
+    private HttpResponse httpResponse;
+    private HttpCookie lwssoCookie;
+    private final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    @Inject
+    private ConnectionSettingsProvider connectionSettingsProvider;
+
+    public HttpResponse loginClient() {
+        ConnectionSettings connectionSettings = connectionSettingsProvider.getConnectionSettings();
+        HttpRequest httpRequest;
+        HttpTransport HTTP_TRANSPORT = new NetHttpTransport();
+        HttpRequestFactory requestFactory = HTTP_TRANSPORT.createRequestFactory();
+        SimpleUserAuthentication authentication = new SimpleUserAuthentication(connectionSettings.getUserName(), connectionSettings.getPassword());
+        ByteArrayContent content = ByteArrayContent.fromString("application/json", authentication.getAuthenticationString());
+
+        try {
+            httpRequest = requestFactory.buildPostRequest(new GenericUrl(connectionSettings.getBaseUrl() + "/authentication/sign_in"),
+                    content);
+            httpResponse = httpRequest.execute();
+        } catch (IOException e) {
+            logger.error(e.getMessage());
+            return null;
+        }
+        return httpResponse;
+
+    }
+
+    private HttpCookie setLwssoCookie() {
+        List<String> strHPSSOCookieCsrf1 = httpResponse.getHeaders().getHeaderStringValues("Set-Cookie");
+
+        for (String strCookie : strHPSSOCookieCsrf1) {
+            List<HttpCookie> cookies = HttpCookie.parse(strCookie);
+            Optional<HttpCookie> lwssoCookieOp = cookies.stream().filter(a -> a.getName().equals("LWSSO_COOKIE_KEY")).findFirst();
+            if (lwssoCookieOp.isPresent()) {
+                lwssoCookie = lwssoCookieOp.get();
+                break;
+            } else {
+                return null;
+            }
+        }
+        return lwssoCookie;
+    }
+
+    /**
+     * @param pictureLink - src link to the image from server
+     * @return
+     */
+    public HttpResponse getImageData(String pictureLink) {
+        if (lwssoCookie == null) {
+            loginClient();
+            lwssoCookie = setLwssoCookie();
+        }
+
+        HttpTransport HTTP_TRANSPORT = new NetHttpTransport();
+        HttpRequestFactory requestFactory = HTTP_TRANSPORT.createRequestFactory();
+        try {
+            HttpRequest httpRequest = requestFactory.buildGetRequest(new GenericUrl(pictureLink));
+            httpRequest.getHeaders().setCookie(lwssoCookie.toString());
+            httpResponse = httpRequest.execute();
+        } catch (IOException e) {
+            logger.error(e.getMessage());
+            return null;
+        }
+        //test http request format when cookie is expired
+        if(httpResponse.getStatusCode() == 403){
```

Comment:
@atothhpe @kdrtibor  question here: Isn't there a case where 408 would happen? 408's description: "The client did not produce a request within the time that the server was prepared to wait. The client MAY repeat the request without modifications at any later time." 
Won't there be a case like this, when having for example a slow connection?

Reply:
In our case the server is Octane which does not have a wait time for a request to happen as far as i know.